kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -763,6 +810,21 @@ def foundry_step_node(
             apr_proof.write_proof()
 
 
+def foundry_get_apr_proof(
+    foundry_root: Path,
+    test: str,
+    bug_report: bool = False,
+) -> APRProof:
+    br = BugReport(Path(f'{test}.bug_report')) if bug_report else None
+    foundry = Foundry(foundry_root, bug_report=br)
+
+    apr_proofs_dir = foundry.out / 'apr_proofs'
+    contract_name, test_name = test.split('.')
+    proof_digest = foundry.proof_digest(contract_name, test_name)
+    apr_proof = APRProof.read_proof(proof_digest, apr_proofs_dir)
+    return apr_proof
```

Comment:
Let's:

- Use `proof` variable name instead of `apr_proof`.
- Make sure it works with both `APRProof` and `APRBMCProof`, by using `Proof.read_proof` instead. We can make an assertion at the end of this function that `isinstance(proof, APRProof)` to make sure it's one of those subclasses if you want/need.

Reply:
I renamed it here, but also for most of the lesser-used commands like `foundry-step-node`, `foundry-remove-node`, etc., those commands were converting `APRBMCProof`s back into `APRProof`s, and I fixed these so I could test the merging feature with BMC proofs.